lib/Wikia/src/Tasks/Tasks/MaintenanceTask.php

```diff
@@ -2,11 +2,50 @@
 
 namespace Wikia\Tasks\Tasks;
 
+use Wikia\Util\Assert;
+
 class MaintenanceTask extends BaseTask {
-	public function run( $script ) {
+
+	/**
+	 * @param string $script
+	 * @throws \Wikia\Util\AssertionException
+	 * @return bool
+	 */
+	public static function validatePath( string $script ) : bool {
+		global $IP;
+
+		$script = realpath( $IP. '/' . $script );
+
+		Assert::true( endsWith( $script, '.php' ), '$script must end with .php' );
+		Assert::true( startsWith( $script, $IP ), 'Script path must be relative to app\'s root' );
```

Comment:
Let's not use exceptions for flow control

Reply:
Why? :) We found assets to be quite useful.